web3/utils/datatypes.py

```diff
@@ -0,0 +1,50 @@
+from functools import (
+    partial,
```

Comment:
Take a look at https://github.com/pipermerriam/ethereum-dev-tactical-manual/blob/master/style-guide.md#avoid-from-x-import-y-for-most-standard-library-imports

The general idea is that seeing `functools.partial` somewhere deep in a module makes it pretty clear that I'm looking at the `partial` function from the standard library `functools`.  However, if all I see is `partial` then I have to go back to the top of the module to find out if I'm actually looking at `functools.partial`.  Makes code slightly easier to parse at the cost of a bit of verbosity.

Reply:
Oh no! its the example in the manual <:‑|

Reply:
Learning a team's code style takes some time and immersion.